build/build.go

```diff
@@ -28,6 +30,25 @@ import (
 	"golang.org/x/tools/go/buildutil"
 )
 
+const (
+	hashDebug = false
+)
+
+var (
+	compilerBinaryHash string
+)
+
+func init() {
+	// We do this here because it will only fail in truly bad situations, i.e.
+	// machine running out of resources. We also panic if there is a problem
+	// because it's unlikely anything else will be useful/work
```

Comment:
Rather than panicking here, would it be useful to fallback to a cache-disabled state?

Maybe that would allow running the gopherjs compiler in unusual environments (such as within a browser?)? (That may already be impossible for other reasons, idk).

Reply:
That's a fair point. I think I'd like to understand that situation a bit better; because a whole load of logic in `build` depends on the OS being available (writing archives etc).

The compiler is run within the browser by the playground, but that does not import `build`, just `compiler`.